.gitlab-ci.yml

```diff
@@ -18,8 +18,7 @@ build_cuda:
         -DCMAKE_BUILD_TYPE=Release
         -DTRACCC_BUILD_TESTING=ON
         -DTRACCC_BUILD_CUDA=ON
-      - cmake --build build
-
+      - cmake --build build -- -j 4
```

Comment:
I set the number of threads to 4 to avoid occupying too much memory space

Reply:
This should be equivalent. The VMs have 4 virtual CPU cores.

Reply:
4 virtual CPU cores maybe equivalent to 8 threads?
I failed the CI tests ~10 times and it passed after changing to 4 threads.

Reply:
The machines have `nproc` reporting 4, I don't think ninja goes beyoned that.

Did you get OOM errors?

Reply:
This is the error I used to get. (It can fail on any other example files - it is almost random)

```
FAILED: examples/run/cpu/CMakeFiles/traccc_seeding_example.dir/seeding_example.cpp.o 
/usr/bin/c++ -DALGEBRA_PLUGINS_INCLUDE_ARRAY -DBOOST_ALL_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DCOVFIE_QUIET -DDETRAY_ALGEBRA_ARRAY
-DDETRAY_ALGEBRA_EIGEN -DDETRAY_ALGEBRA_VC -DDETRAY_CUSTOM_SCALARTYPE=float -DJSON_DIAGNOSTICS=0 -DJSON_USE_IMPLICIT_CONVERSIONS=1
-DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CUDA -DTHRUST_HOST_SYSTEM=THRUST_HOST_SYSTEM_CPP -DTRACCC_CUSTOM_SCALARTYPE=float -DVECMEM_DEBUG_MSG_LVL=0
-DVECMEM_HAVE_PMR_MEMORY_RESOURCE -DVECMEM_SOURCE_DIR_LENGTH=27 -I/builds/acts/ci-bridge/build/_deps/vecmem-build/core/CMakeFiles -I/builds/acts/ci-
bridge/build/_deps/vecmem-src/core/include -I/builds/acts/ci-bridge/src/core/include -I/builds/acts/ci-bridge/build/_deps/detray-src/core/include
-I/builds/acts/ci-bridge/build/_deps/detray-build/core/CMakeFiles -I/builds/acts/ci-bridge/src/plugins/algebra/array/include -I/builds/acts/ci-
bridge/build/_deps/algebraplugins-src/frontend/array_cmath/include -I/builds/acts/ci-bridge/build/_deps/algebraplugins-src/common/include
-I/builds/acts/ci-bridge/build/_deps/algebraplugins-src/storage/array/include -I/builds/acts/ci-bridge/build/_deps/algebraplugins-
src/math/cmath/include -I/builds/acts/ci-bridge/build/_deps/algebraplugins-src/math/common/include -I/builds/acts/ci-bridge/build/_deps/detray-
src/plugins/algebra/array/include -I/builds/acts/ci-bridge/src/plugins/algebra/vecmem/include -I/builds/acts/ci-bridge/build/_deps/algebraplugins-
src/frontend/vecmem_cmath/include -I/builds/acts/ci-bridge/build/_deps/algebraplugins-src/storage/vecmem/include -I/builds/acts/ci-
bridge/src/plugins/algebra/eigen/include -I/builds/acts/ci-bridge/build/_deps/algebraplugins-src/frontend/eigen_eigen/include -I/builds/acts/ci-
bridge/build/_deps/algebraplugins-src/storage/eigen/include -I/builds/acts/ci-bridge/build/_deps/algebraplugins-src/math/eigen/include
-I/builds/acts/ci-bridge/build/_deps/detray-src/plugins/algebra/eigen/include -I/builds/acts/ci-bridge/src/plugins/algebra/vc/include
-I/builds/acts/ci-bridge/build/_deps/algebraplugins-src/frontend/vc_vc/include -I/builds/acts/ci-bridge/build/_deps/algebraplugins-
src/storage/vc/include -I/builds/acts/ci-bridge/build/_deps/algebraplugins-src/math/vc/include -I/builds/acts/ci-bridge/build/_deps/detray-
src/plugins/algebra/vc/include -I/builds/acts/ci-bridge/build/_deps/algebraplugins-src/frontend/vc_cmath/include -I/builds/acts/ci-
bridge/src/io/include -I/builds/acts/ci-bridge/build/_deps/acts-src/Core/include -I/builds/acts/ci-bridge/build/_deps/acts-build/Core
-I/builds/acts/ci-bridge/src/performance/include -I/builds/acts/ci-bridge/build/_deps/covfie-src/lib/core -I/builds/acts/ci-
bridge/src/examples/options/include -I/builds/acts/ci-bridge/build/_deps/detray-src/utils/include -I/builds/acts/ci-bridge/build/_deps/detray-
src/io/include -I/builds/acts/ci-bridge/build/_deps/dfelibs-src -I/builds/acts/ci-bridge/build/_deps/nlohmann_json-src/single_include -isystem
/builds/acts/ci-bridge/build/_deps/eigen3-src -isystem /builds/acts/ci-bridge/build/_deps/thrust-src -isystem /builds/acts/ci-
bridge/build/_deps/thrust-src/dependencies/cub -isystem /builds/acts/ci-bridge/build/_deps/vc-src -Wall -Wextra -Wshadow -Wunused-local-typedefs
-pedantic -O3 -DNDEBUG -std=c++17 -MD -MT examples/run/cpu/CMakeFiles/traccc_seeding_example.dir/seeding_example.cpp.o -MF
examples/run/cpu/CMakeFiles/traccc_seeding_example.dir/seeding_example.cpp.o.d -o
examples/run/cpu/CMakeFiles/traccc_seeding_example.dir/seeding_example.cpp.o -c /builds/acts/ci-bridge/src/examples/run/cpu/seeding_example.cpp
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
```

I could find some threads that say `c++: fatal error: Killed signal terminated program cc1plus` is from OOM

Reply:
Yeah that's an OOM. Anyway, no harm in keeping it at explicit `-j4` in any case.

Reply:
If memory is an issue, just hardcode 1 or 2 threads. 🤔 It's not great, but in the end the build on GitLab is not that super long in the end.